### PR TITLE
update TensorRT EP to use TensorRT 8.2

### PR DESCRIPTION
**Pull request by jywu-msft**
update TensorRT EP to use the latest TensorRT version 8.2 GA (8.2.1.8)

-update onnx-tensorrt parser submodule to 8.2-GA branch
-disable known unit test failures for TensorRT 8.2
-update CI and packaging pipelines to use TensorRT 8.2

### CODE REVIEW

**Comment by jywu-msft** [comment]
@oliviajain, can you review the changes made to linux-gpu-tensorrt-ci-perf-pipeline.yml and try it out?

File: tools/ci_build/github/azure-pipelines/linux-gpu-tensorrt-ci-perf-pipeline.yml [hunk 1]
```diff
@@ -23,8 +23,9 @@ parameters:
 - name: TrtVersion
   displayName: TensorRT Version
   type: string
-  default: 8.0.1.6
+  default: 8.2.1.8
   values:
+  - 8.2.1.8
```
Comment (oliviajain) [inline tools/ci_build/github/azure-pipelines/linux-gpu-tensorrt-ci-perf-pipeline.yml:28]: Pipeline file deprecated, change not needed.

Comment (jywu-msft) [reply tools/ci_build/github/azure-pipelines/linux-gpu-tensorrt-ci-perf-pipeline.yml]: i deleted the file.

**Comment by stevenlix** [comment]
Are we still going to use Nvidia TensorRT container moving forward? If not, should Dockerfile.ubuntu_tensorrt be removed?

**Comment by jywu-msft** [comment]
> Are we still going to use Nvidia TensorRT container moving forward? If not, should Dockerfile.ubuntu_tensorrt be removed?

it's probably useful to have the dockerfile around for testing on ubuntu
we probably should have a separate pipeline for testing it on demand. can be done as a separate task.

**Review by stevenlix — APPROVED** [review]

**Pull request merged**